### PR TITLE
add name to post

### PR DESCRIPTION
**Pull request by adoolaeghe**

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <div id='post-<%= post.id %>' class="post">
     <div class="post__head">
       <span class="post__head__user">
-        <%= post.id  %>
+        <%= post.user.first_name %> <%=post.user.last_name  %>
```
Comment (dearshrewdwit) [inline app/views/posts/index.html.erb]: Is there a way to make this code simpler? What is the object that you need for the names? How could you pass it in?

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (adoolaeghe) [reply app/views/posts/index.html.erb]: <%= post.user.first_name + " " + post.user.last_name  %> ???

File: app/models/post.rb [hunk 2]
```diff
@@ -20,4 +20,11 @@ def summary
     return message unless message.length > MAX_SUMMARY_LENGTH
     message[0...MAX_SUMMARY_LENGTH] + "..."
   end
+
+  def first_name
+    return user.first_name
```
Comment (dearshrewdwit) [inline app/models/post.rb]: How does implicit return in Ruby work?

File: app/models/post.rb [hunk 3]
```diff
@@ -20,4 +20,11 @@ def summary
...
+  end
+  def last_name
+    return user.first_name
```
Comment (dearshrewdwit) [inline app/models/post.rb]: How does implicit return in Ruby work?

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -2,7 +2,7 @@
   <div id='post-<%= post.id %>' class="post">
     <div class="post__head">
       <span class="post__head__user">
-        <%= post.id  %>
+        <%= post.first_name + " " + post.last_name  %>
```
Comment (dearshrewdwit) [inline app/views/posts/index.html.erb]: If this is the intended output, maybe use one method on the model that returns the full name?

And TEST it 🎉

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: app/models/post.rb [hunk 5]
```diff
@@ -20,4 +20,8 @@ def summary
     return message unless message.length > MAX_SUMMARY_LENGTH
     message[0...MAX_SUMMARY_LENGTH] + "..."
   end
+
+  def user_name
+    user.first_name + user.last_name
```
Comment (dearshrewdwit) [inline app/models/post.rb]: where is the unit test for this?

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

**Comment by dearshrewdwit** [comment]
remember to write a description of the PR

File: app/models/post.rb [hunk 6]
```diff
@@ -20,4 +20,8 @@ def summary
     return message unless message.length > MAX_SUMMARY_LENGTH
     message[0...MAX_SUMMARY_LENGTH] + "..."
   end
+
+  def user_name
+    "#{user.first_name} #{user.last_name}"
```
Comment (dearshrewdwit) [inline app/models/post.rb:25]: 👍

**Review by dearshrewdwit — DISMISSED** [review]

**Comment by dearshrewdwit** [comment]
@adoolaeghe please remember to fill out the description of your PRs

**Comment by tobold** [comment]
Please don't merge this until the database has been migrated to remove NULL:
https://github.com/makersacademy/acebook-july2017/pull/28

**Review by tobold — APPROVED** [review]
simply rebasing, old review still applies

**Pull request merged**